### PR TITLE
hammer: crush/CrushTester: do not pass a transient string to cmd_args 

### PR DESCRIPTION
**Pull request by choury**
I got the orgin params with strace like this:
 ["timeout", "5", "crushtool", "-i", "-", "--test", "--check", "49", "--min-x", "1", "--max-x", "50", "--ruleset", ""]

This will cause  creating  pool fail:
Error EINVAL: error running crushmap through crushtool: (1) Operation not permitted

ref #13357

Fixes: http://tracker.ceph.com/issues/18890
Signed-off-by: Wei Zhou <chouryzhou@tencent.com>

Conflicts:
    src/crush/CrushTester.cc: The hammer code was putting the char* on vector of char*, whereas on master SubProcess:add_cmd_arg is taking a copy onto a vector of std::string.

### CODE REVIEW

**Comment by tchaikov** [comment]
could you add a `Conflicts` section in the commit message explaining why this fix is not cherry-picked from master. and also, could you revise the commit message and repush, so it looks like:
```
crush/CrushTester: do not pass a transient string to cmd_args

I got the orgin params with strace like this:
 ["timeout", "5", "crushtool", "-i", "-", "--test", "--check", "49", "--min-x", "1", "--max-x", "50", "--ruleset", ""]

This will cause  creating  pool fail:
Error EINVAL: error running crushmap through crushtool: (1) Operation not permitted

Fixes: http://tracker.ceph.com/issues/18890
Signed-off-by: Wei Zhou <chouryzhou@tencent.com>
Conflicts:
    src/crush/CrushTester.cc: <the reason why this commit is not cherry-picked from master>
```

thanks!

**Comment by choury** [comment]
@tchaikov fixed

File: src/crush/CrushTester.cc [hunk 1]
```diff
@@ -364,6 +364,7 @@ int CrushTester::test_with_crushtool(const string& crushtool,
 {
   string timeout_string = stringify(timeout);
   string opt_max_id = stringify(max_id);
+  string opt_ruleset = stringify(ruleset);
```
Comment (amitkumar50) [inline src/crush/CrushTester.cc]: @choury (NIT) Title of pull Req < 50 characters
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

Comment (choury) [inline src/crush/CrushTester.cc]: Change title to "hammer: crush: don’t use a transient string" ?

**Comment by smithfarm** [comment]
@choury The hammer series will be declared End-of-Life after luminous is released (i.e. very soon). Unless a very serious bug is found, there will probably not be another hammer point release.

**Comment by choury** [comment]
@smithfarm That's OK, we've fixed it on our branch. Just want to do something about community feedback

File: src/crush/CrushTester.cc (hunk 1 above)
Comment (smithfarm) [inline src/crush/CrushTester.cc]: @choury You can change the title of the pull request if you like, but please do not touch the commit message. This is a backport, so the commit message is derived from the master commit, with any changes described in the Conflicts section. And @choury you did this very well already - your commit message is excellent.

@amitkumar50 Please read our backporting guidelines at http://tracker.ceph.com/projects/ceph-releases/wiki - I believe your reviews are well-intentioned, but for backports we want as few changes (compared to master) as possible. This is explained in the guidelines.

Bottom line: all changes must go into master first, if possible, because master is where development takes place.

OK?

**Comment by smithfarm** [comment]
Hammer is EOL.
